### PR TITLE
CRNCC-2286 cancel away from editing dependent

### PR DESCRIPTION
**Pull request by Tkirkwood**
model Id was not passed in method so the dependant question did not have it to link back to in both the cancel button as well as the breadcrumbs.

https://nihr.atlassian.net/browse/CRNCC-2286

### CODE REVIEW

**Review by paul-tennyson — APPROVED** [review]

File: BPOR/BPOR.Rms/Controllers/ResearcherController.cs [hunk 1]
```diff
@@ -468,7 +468,7 @@ public async Task<IActionResult> Edit(int id, int field)
     [HttpPost]
     [ValidateAntiForgeryToken]
     public async Task<IActionResult> Edit(int id,
-        [Bind("ShortName,ChiefInvestigator,StudySponsors,CPMSId,PortfolioSubmissionStatus,OutcomeOfSubmission," +
+        [Bind("ShortName,ChiefInvestigator,StudySponsors,CPMSId,PortfolioSubmissionStatus,OutcomeOfSubmission,Id," +
```
Comment (con-nihr) [inline BPOR/BPOR.Rms/Controllers/ResearcherController.cs:471]: If we need to `Bind` it is probably a sign the model is too complicated.
The model should only include properties that we want to bind anyway.

`Id` is already passed in as a route parameter so arguably we should have a few different models depending on what is needed.

**Review by con-nihr — CHANGES_REQUESTED** [review]
I'll support some of these changes to restructure the model as needed.

File: BPOR/BPOR.Rms/Controllers/ResearcherController.cs (hunk 1 above)
Comment (Tkirkwood) [reply BPOR/BPOR.Rms/Controllers/ResearcherController.cs:471]: I went back over the method and noticed I overlooked a line near the end of the code
`model.Id = id;` 
this is not reached when going through a dependancy question as that returns much further up.
would it be better to remove the bound id and move this line nearer the start of the method, or is a restructure for the model still the better option?

Comment (con-nihr) [reply BPOR/BPOR.Rms/Controllers/ResearcherController.cs:471]: I think we could benefit from a slight larger restructuring of things in this area, but if we want to keep things compact to close out this fix I would be fine with moving `model.Id = id;` right up to the first line of the method.

File: BPOR/BPOR.Rms/Controllers/ResearcherController.cs [hunk 2]
```diff
@@ -479,6 +479,7 @@ public async Task<IActionResult> Edit(int id,
 
         ValidateMandatoryFields(model);
 
+        model.Id = id;
```
Comment (con-nihr) [inline BPOR/BPOR.Rms/Controllers/ResearcherController.cs:482]: I would literally put this as the first line.

There may be future changes to `ValidateMandatoryFields()` that might expect `model.Id` to be populated.

**Review by con-nihr — CHANGES_REQUESTED** [review]

**Review by con-nihr — APPROVED** [review]

**Pull request merged**